source/common/common/assert.h

```diff
@@ -13,6 +16,46 @@ class ActionRegistration {
 };
 using ActionRegistrationPtr = std::unique_ptr<ActionRegistration>;
 
+/*
+ * EnvoyBugStackTrace captures and writes the stack trace to Envoy Bug
+ * to assist with getting additional context for reports
+ */
+class EnvoyBugStackTrace : Logger::Loggable<Logger::Id::envoy_bug> {
+public:
+  EnvoyBugStackTrace() = default;
+  /*
+   * Capture the stack trace
+   * Skip count is one as to skip the last call which is capture()
+   */
+  void capture() {
+    stack_depth_ = absl::GetStackTrace(stack_trace_, MaxStackDepth, /* skip_count = */ 1);
+  }
+
+  /*
+   * Logs each row of the captured stack into the envoy_bug log
+   */
+  void logStackTrace() {
+    ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::envoy_bug), error,
+                        "stacktrace for envoy bug");
+    for (int i = 0; i < stack_depth_; ++i) {
+      char out[1024];
+      const bool success = absl::Symbolize(stack_trace_[i], out, sizeof(out));
+      if (success) {
```

Comment:
Just for clarity. Are you suggesting that it should always log a certain number of items, even if the stack trace doesn't have that many items. So say the current max is 16. If it only goes to say 10 calls, should line 11-16 just be unknown? Or are you suggesting just altering the !success to be "Unknown" instead of blank?

Reply:
printing `"#{} {} [{}]", i, out, stack_trace_[i]) ` and have it use "UNKNOWN if out failed

Reply:
Understood. Done